### PR TITLE
[OpenAI] Regenerated with Multipart

### PR DESCRIPTION
**Pull request by mssfang**
- Hid `getAudioTranscriptionAsPlainTextWithResponse` and `getAudioTranslationAsPlainTextWithResponse` methods from `OpenAIClient` and `OpenAIAsyncClient` classes.
- Made `ImageGeneration` constructor as private.
- Made `ImageGenerationData` constructor as private and removed setter methods.

- Added new properties `ImageGenerationContentFilterResults contentFilterResults` and `ImageGenerationPromptFilterResults promptFilterResults`
  in `ImageGenerationData` class to support filtering results.
- Added new property `suffix` in `CompletionsOptions` class to support suffix for completions.

### CODE REVIEW

File: sdk/openai/azure-ai-openai/CHANGELOG.md [hunk 1]
```diff
@@ -29,19 +25,27 @@
 - Replaced `String` property `name` by `ChatCompletionsFunctionToolSelection` property `function` in `ChatCompletionsNamedFunctionToolSelection`
 - Made `embeddingDependency` as a required parameter in `AzureCosmosDBChatExtensionParameters` and `PineconeChatExtensionParameters` class, and removed setter method.
 - Removed `vectorFields` and `imageVectorFields` from `PineconeFieldMappingOptions` class, and made `contentField` as required parameter.
+- Hid `getAudioTranscriptionAsPlainTextWithResponse` and `getAudioTranslationAsPlainTextWithResponse` methods from `OpenAIClient` and `OpenAIAsyncClient` classes.
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/CHANGELOG.md:28]: We should just mentioned that this is removed as hiding the API is equivalent to removing the API from the library users' perspective.

```suggestion
- Removed `getAudioTranscriptionAsPlainTextWithResponse` and `getAudioTranslationAsPlainTextWithResponse` methods from `OpenAIClient` and `OpenAIAsyncClient` classes.
```

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java [hunk 2]
```diff
@@ -665,28 +665,34 @@ public Flux<Completions> getCompletionsStream(String deploymentOrModelName, Comp
...
+     * Completions support a wide variety of tasks and generate text that continues from or "completes"
+     * provided prompt data on successful completion of {@link Mono}.
      */
+    @Generated
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<ChatCompletions> getChatCompletions(String deploymentOrModelName,
         ChatCompletionsOptions chatCompletionsOptions) {
+        // Generated convenience method for getChatCompletionsWithResponse
         RequestOptions requestOptions = new RequestOptions();
-        return getChatCompletionsWithResponse(deploymentOrModelName, chatCompletionsOptions, requestOptions)
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:688]: Did we remove the strongly typed "WithResponse"  method?

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java [hunk 3]
```diff
@@ -1115,8 +1123,10 @@ Response<BinaryData> getAudioTranslationAsResponseObjectWithResponse(String depl
      */
     @Generated
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Response<BinaryData> getAudioTranslationAsPlainTextWithResponse(String deploymentOrModelName,
+    Response<BinaryData> getAudioTranslationAsPlainTextWithResponse(String deploymentOrModelName,
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java:1126]: When manually changing the method, you should also remove `@Generated` annotation.

File: sdk/openai/azure-ai-openai/CHANGELOG.md [hunk 4]
```diff
@@ -29,19 +25,27 @@
...
+- Made `ImageGeneration` constructor as private.
+- Made `ImageGenerationData` constructor as private and removed setter methods.
 
 ### Bugs Fixed
 
 - Fixed `ChatRequestUserMessage` deserialization issue. [#38183](https://github.com/Azure/azure-sdk-for-java/issues/38183)
 
 ### Other Changes
 
+- Dropped service API version support for `2023-08-01-preview`, `2023-09-01-preview` and `2023-12-01-preview`.
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/openai/azure-ai-openai/CHANGELOG.md:38]: Yes. CI and Live tests passed

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java (hunk 3 above)
Comment (mssfang) [reply sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java:1126]: it is automatically. Weidong fixes the issue in https://github.com/Azure/azure-rest-api-specs/commit/da72197d2ab5d7b0d2756976a3cf2927fc60adc4

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java (hunk 2 above)
Comment (mssfang) [reply sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:688]: We still have the strongly typed method in https://github.com/Azure/azure-sdk-for-java/pull/38975/files/26d701f98c494e981148e9e2fd8643d3f885bc48#diff-ecfcbaef21e9bc3878272dbf8e46da30de1ce4b991418e6b86af1f1cbf43fe2fL552

**Review by srnagar — APPROVED** [review]

**Pull request merged**